### PR TITLE
Add support for degrees Celsius to FITS format

### LINKED ISSUE #14033: Degrees Celsius should be supported by FITS units
### Description

The FITS standards says that units should follow IAU recommendations. These note that:

> The degree Celsius (`°C` in the original PDF, `oC` in the web page???) is used in specifying temperature for meteorological purposes, but otherwise the kelvin (K) should be used.

However, astropy does not support `u.deg_C` for fits:

```
import astropy.units as u

u.deg_C.to_string("fits") # exception
```

### Additional context
See 
* https://www.iau.org/publications/proceedings_rules/units/
* https://www.iau.org/static/publications/stylemanual1989.pdf

### PR DESCRIPTION
**Pull request by maxnoe**
### Description

Fixes #14033 

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label. Codestyle issues can be fixed by the [bot](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit).
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: astropy/units/format/fits.py
Comment (mhvk) [inline astropy/units/format/fits.py:32]: Ah, that is such a simple solution, to just add them up front! But maybe add a comment along the lines of "Start with some units for which the standard unit name is different."

File: astropy/units/si.py
Comment (mhvk) [inline astropy/units/si.py:255]: Sometimes I am so pleasantly surprised at how the machinery just extends easily!

File: docs/changes/units/14042.feature.rst
Comment (mhvk) [inline docs/changes/units/14042.feature.rst:2]: I'd swap "Celsius" and "deg C" to make clear the former is preferred, and perhaps be explicit about that: "Note that "deg C" is only provided for compatibility with existing FITS files, as it does not conform to the normal unit standard, where this should be read as "degree * Coloulomb". Indeed, compound units ..."

**Review by mhvk — APPROVED** [review]
It is very nice to see that this was actually easier than I thought it would be!

Will approve now, since my comments are really just suggestions for an additional comment and a slight clarification in the release note.

**Comment by maxnoe** [comment]
Comments addressed

**Comment by maxnoe** [comment]
@mhvk merge? Or do we need a second review?

**Comment by mhvk** [comment]
I think it is ready to go in, but I would be tempted to squash the commits and wrap the changelog fragment a bit more nicely. Am happy to do that.

**Comment by maxnoe** [comment]
done

**Review by mhvk — APPROVED** [review]
Super, let's get this in!

**Pull request merged**
